Java8

FireBurn opened the pull request:
This gets OpenAM building for me with Oracle JDK 8

k-tamura committed:
Fix for OPENAM-1629, review CR-3902: OPENAM-1629: Return value of String.trim() is not used

aldaris committed:
Fix for OPENAM-3766 - review: CR-3978
Updated the TreeSearchIndexTest case to compare sets correctly.

aldaris committed:
OPENAM-141 CR-4282 Remove deprecated JDK API calls from openam-shared

FireBurn commented:
I think I've stumbled onto this issue when compiling with Java 8 and trying to run with IBM's Java 7

https://gist.github.com/AlainODea/1375759b8720a3f9f094

I'll look into this tomorrow if you're interested in getting this work merged

aldaris commented on openam-shared/src/main/java/com/iplanet/security/util/ObjectIdentifier.java:
```diff
@@ -25,6 +25,10 @@
  * $Id: ObjectIdentifier.java,v 1.2 2008/06/25 05:52:44 qcheng Exp $
```
First commit looks good.

aldaris commented on openam-core/src/test/java/org/forgerock/openam/entitlement/indextree/TreeSearchIndexTest.java:
```diff
@@ -24,6 +24,7 @@
 import java.util.HashSet;
```
Second commit looks good.

aldaris commented on openam-authentication/openam-auth-cert/src/main/java/com/sun/identity/authentication/modules/cert/Cert.java:
```diff
@@ -24,10 +24,7 @@
  *
```
Third commit looks good.

aldaris commented on openam-core/src/test/java/org/forgerock/openam/cts/adapters/SAMLAdapterTest.java:
```diff
@@ -100,8 +100,5 @@ public void shouldNotStoreSecondaryKeyIfNull() {
         Token token = adapter.toToken(samlToken);
-
-        // Then
-        assertThat(token.getValue(SAMLTokenField.SECONDARY_KEY.getField())).isNull();
```
Looks to be unrelated change

FireBurn replied:
Unfortunately it wasn't unrelated, the code didn't compile I was working on a fix and it looks like I forgot about it :$

FireBurn replied:
Fixed by pulling in OPENAM-4511

aldaris commented on openam-core/src/test/java/org/forgerock/openam/cts/adapters/SessionAdapterTest.java:
```diff
@@ -182,7 +184,7 @@ public void shouldAssignSessionID() {
 
         // Then
-        assertThat(token.getValue(SessionTokenField.SESSION_ID.getField())).isEqualTo(sessionId);
+        assertThat(StringUtils.isEqualTo(token.getValue(SessionTokenField.SESSION_ID.getField()).toString(),sessionId));
```
Also not related, and it even makes the test useless

aldaris commented on openam-distribution/openam-distribution-ssoadmintools/pom.xml:
```diff
@@ -166,6 +166,21 @@
+
+        <dependency>
+            <groupId>xerces-J</groupId>
+            <artifactId>xml-apis</artifactId>
```
just leaving this as a sidenote: in newer versions xerces has been removed and instead we are using the XML parser provided by the JDK

FireBurn replied:
I'll look at the commits you mentioned and see if there's a better way

aldaris commented on openam-shared/src/main/java/org/forgerock/openam/utils/StringUtils.java:
```diff
@@ -51,6 +58,22 @@ public static String insertContent(String original, int position, String content
+     * @param s2 second string to compare.
+     * @return true if strings are equal or they are both null.
+     */
+    public static boolean isEqualTo(String s1, String s2) {
```
probably just as unrelated

FireBurn replied:
I'll pop that into it's own commit

FireBurn replied:
Not required since pulling in OPENAM-4511

aldaris commented on openam-documentation/openam-doc-source/src/main/docbkx/admin-guide/chap-troubleshooting.xml:
```diff
@@ -83,13 +83,13 @@ $ jar cf ../openam.war *</screen>
      started, use the <command>ps</command> command to check. In the following
      example, the web container is <literal>tomcat</literal>.</para>
      <screen>$ ps -ef | grep tomcat | grep Xm
-... -Xmx1024m -XX:MaxPermSize=256m ...</screen>
```
these settings are still relevant when people are using 1.7. As long as 1.7 is the compiler source/target version, probably these should stick around.

FireBurn replied:
I'll see about parameterising that too

aldaris commented on pom.xml:
```diff
@@ -1749,6 +1758,7 @@
                     <encoding>utf-8</encoding>
                     <source>${java.source.version}</source>
                     <additionalJOption>${javadoc.options}</additionalJOption>
+                    <additionalparam>${javadoc.opts}</additionalparam>
```
have you verified that this works with 1.7 as well, will the plugin error out because of the empty string value on 1.7?

FireBurn replied:
It does work on both Java7 & Java8 as the compiler. That's what I changed in 
d6c6186 when I realised I'd broken the compile with Java7

aldaris reviewed (changes requested):
This is mostly good.
FYI The original Java8 migration work required:
OPENAM-3766
OPENAM-3767
OPENAM-4170
OPENAM-4510
OPENAM-4511
OPENDJ-1142
OPENAM-141

FireBurn committed:
OPENAM-4511 CR-4605 Resolve test compilation issues on JDK8

Based on:

commit e831d0765b961c16c5484635f324ef778a160a05
Author: Peter Major <majorpetya@gmail.com>
Date:   Mon Sep 22 20:07:42 2014 +0000

FireBurn committed:
Fix for OPENAM-2673, review CR-3256

Make use of Apache Santuario for XML Signature API.

Based on:

commit 272ac8a1a482b3baeff7293aac5de828cfd1ee69
Author: Mark de Reeper <mark.dereeper@forgerock.com>
Date:   Tue Apr 15 04:07:56 2014 +0000

FireBurn committed:
Minor changes to stop Java 8 erroring out

This puts java.surefire.options and javadoc.options into profiles for
JDK 1.7 and JDK 1.8+ so MaxPermSize can be ommited on newer JDKs

It also adds a javadoc.opts parameter so -Xdoclint:none can be passed

FireBurn committed:
Use https for maven repos

FireBurn committed:
Switch from log4j to reload4j

The apache project log4j 1 is no longer being updated, the origional
author has created reload4j to make some fixes

More details can be found here:

https://reload4j.qos.ch/

FireBurn committed:
Add required binaries

FireBurn committed:
Use AES_256/GCM/NoPadding
